src/enochecker/enochecker.py

```diff
@@ -448,16 +485,45 @@ def havoc(self) -> None:
         pass
 
     @abstractmethod
-    def exploit(self) -> None:
+    def exploit(self) -> str:
         """
         Use this method strictly for testing purposes.
 
         Will hopefully not be called during the actual CTF.
 
-        :raises: EnoException on Error
+        :return: The flag found during the execution of this exploit
+        :raises: EnoException on Error or if the flag was not found
         """
         pass
 
+    def search_flag(self, data: str) -> Optional[str]:
```

Comment:
This could be moved to `enochecker_core`

Reply:
at the moment `enochecker_core` contains only the dataclasses implementing the specification and does not contain any utility functions

Reply:
Then that's the perfect time to change this? 😀  Any reason not to do it?

Reply:
well, in this case it's a 5 line function that is also embedded into the enochecker-class, meaning refactoring this woud lead to a function with more arguments instead. I am not opposed to this in general, but it does not really seem like that big of an issue to have something similar in `enochecker_async`. If we decide to refactor this, we should also consider which other utils can be moved to the `enochecker_core` package